sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -687,20 +699,23 @@ object DDLUtils {
 
   /**
    * If the command ALTER VIEW is to alter a table or ALTER TABLE is to alter a view,
-   * issue an exception [[AnalysisException]].
+   * issue an exception [[AnalysisException]]. Temporary views can be altered by both
+   * ALTER VIEW and ALTER TABLE commands.
```

Comment:
and we can remove this trick if we decide to not allow temporary views to be created by `CREATE TEMPORARY TABLE`.